src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java

```diff
@@ -48,12 +47,16 @@ public class NotificationUtils {
     /**
      * @param otpUser  target user
      * @param textTemplate  template to use for email in text format
-     * @param templateData          template data
+     * @param templateData  template data
      */
     public static String sendPush(OtpUser otpUser, String textTemplate, Object templateData) {
         try {
             String body = TemplateUtils.renderTemplate(textTemplate, templateData);
-            return sendPush(otpUser.email, body);
+            String toUser = otpUser.email;
+            // Calls the `get` endpoint, the only reliable way to get number of push notification devices,
+            // as the `publish` endpoint returns success for zero devices and/or if devices turn them off.
+            otpUser.pushDevices = getPushInfo(toUser);
+            return otpUser.pushDevices > 0 ? sendPush(toUser, body) : "OK";
```

Comment:
Does this update `pushDevices` in Mongo? I don't see any other places where that happens.